### ISSUE TITLE
Data writer channel capacity is too small

### ISSUE THREAD

**Issue by timsaucer** [jhelovuo/RustDDS#33]
In ` src/dds/pubsub.rs` from commit #3117e31b1f1ecbc24383357b02b7372c26e1f629 the size of the mio channels are set to 4.

```
    // Data samples from DataWriter to HistoryCache
    let (dwcc_upload, hccc_download) = mio_channel::sync_channel::<WriterCommand>(4);
    // Status repoerts back from Writer to DataWriter
    let (status_sender, status_receiver) = mio_channel::sync_channel(4);
```

It was previously set to 100. Is there a reason for this to be limited so small? I am trying to create more than 4 subscribers and this is causing an error:

```
2021-04-12 20:01:40,834 WARN  [rustdds::dds::with_key::datawriter] Failed to write new data. Full
2021-04-12 20:01:40,834 ERROR [rustdds::discovery::discovery] Unable to write new topic info.
```

**Comment by jhelovuo** [comment]
The previous value of 100 was judged to be excessively large, as each channel slot allocates memory.

But I do not immediately see how creating more than four subscribers causes these channels to become full. Can you post a minimal example application code that triggers this error?

**Comment by timsaucer** [comment]
Attached is a minimal example of creating 5 readers on one subscriber. Here is the info level output:

```
2021-04-17 07:52:26,767 INFO  [rustdds::dds::participant] ParticipantId 0 selected.
2021-04-17 07:52:26,771 INFO  [rustdds::dds::participant] Discovery started. Participant constructed.
2021-04-17 07:52:26,772 INFO  [rustdds::dds::dp_event_loop] update_participant - begin for GuidPrefix { entityKey: [25, 223, 213, 68, 188, 11, 74, 99, 147, 84, 44, 15] }
2021-04-17 07:52:26,772 INFO  [rustdds::dds::dp_event_loop] update_participant - finished for GuidPrefix { entityKey: [25, 223, 213, 68, 188, 11, 74, 99, 147, 84, 44, 15] }
2021-04-17 07:52:28,770 WARN  [rustdds::dds::with_key::datawriter] Failed to write new data. Full
2021-04-17 07:52:28,770 ERROR [rustdds::discovery::discovery] Unable to write new readers info. OutOfResources
```

Attached is also the output when setting SimpleLogger to trace.

Please let me know if I'm using it not as expected or can provide any more help. Thank you!

[trace_output_multiple_readers.txt](https://github.com/jhelovuo/RustDDS/files/6329175/trace_output_multiple_readers.txt)

[multiple_topics.txt](https://github.com/jhelovuo/RustDDS/files/6329176/multiple_topics.txt)

**Comment by jhelovuo** [comment]
Yes, this was a genuine bug (or messing feature).

Does the latest Release 0.4.1 work for you?

**Comment by timsaucer** [comment]
Yes, it does! Thank you for your quick response. I really appreciate it.

Corrected by commit 6aa747e135388e419ad4b2e73948506126d62ec0
